mnist_test.go

```diff
@@ -24,8 +25,7 @@ func ExampleMNIST() {
 	})
 	trainLoader := torch.NewDataLoader(dataset, 8)
 	for trainLoader.Scan() {
-		torch.GC()
```

Comment:
I moved `torch.GC()` to the inside of `Scan`, because `Scan` function allocated  Tensor memory, we should call it before new  Tensor memory. Users can also write one less line code.